src/liveMarkbookCalculator.js

```diff
@@ -163,14 +256,26 @@ const makeMarkbookEditable = () => {
         if (span) {
             $(span).bind('click', function () {
                 input.show();
+                input.focus();
                 calculateMarks();
                 $(this).remove();
             });
         }
-        $(input).bind('input', function () {
-            calculateMarks();
-            $(this).parent().css('background-color', '#ffe499'); // Change color of cell to indicate it was modified
-        });
+        
+        const margin = $(this).parent().find('td:first > span:first')[0].style['margin-left']; // determines if the row is for an assignment, section, or unit
+        const isMarkColumn = $(this).nextAll().length === 3; // the mark column has 3 cells after it
+
+        // only make assignment marks editable; section and unit marks are dependent on assignments, so their input fields are disabled
+        if (margin !== '40px' && isMarkColumn) {
+            $(input).prop('disabled', true); // to maintain compatibility with other functions, the input is disabled rather than completely removed
+            $(input).css('cursor', 'text'); // give the appearance of regular text
```

Comment:
This causes an issue with malformed markbooks that don't have the `assignment` or bottom layer, they would no longer be editable

Reply:
Do you mean cases where units or sections don't have any assignments?

Reply:
Yes